src/acb/test/t-sqrts.c

```diff
@@ -0,0 +1,91 @@
+/*
+    Copyright (C) 2023 Jean Kieffer
+
+    This file is part of Arb.
+
+    Arb is free software: you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.  See <http://www.gnu.org/licenses/>.
+*/
+
+#include "acb.h"
+
+int main(void)
+{
+    slong iter;
+    flint_rand_t state;
+
+    flint_printf("sqrts....");
+    fflush(stdout);
+
+    flint_randinit(state);
+
+    /* check sqrt overlaps one of the results, y1 = -y2, no precision loss */
```

Comment:
This test code doesn't seem to cover the hard case. Maybe generate a random z with acb_randtest, compute w = z^2, and check that sqrts(w) contains both z and -z ?